name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/FidelityPDFExtractor.java

```diff
@@ -0,0 +1,216 @@
+                        .assign((t, v) -> {
+                            String date = v.get("date");
+                            // convert month to camel case
+                            if (Character.isLetter(date.charAt(0)))
+                            {
+                                date = date.charAt(0) + date.substring(1, 3).toLowerCase() + date.substring(3);
+                            }
+                            t.setDate(asDate(date));
+                        })
+
+                        .section("shares") //
+                        .match("^YOU SOLD (?<shares>[\\.,\\d]+) AT.*$") //
+                        .assign((t, v) -> t.setShares(asShares(v.get("shares"))))
+
+                        .section("gross", "tickerSymbol") //
+                        .match("SECURITY DESCRIPTION SYMBOL: (?<tickerSymbol>[A-Z]{2,5}) Sale Proceeds\\s+\\$(?<gross>[\\.,\\d]+)$") //
+                        .assign((t, v) -> {
+                            t.setCurrencyCode(asCurrencyCode("USD"));
+                            v.put("currency", "USD");
+                            t.setSecurity(getOrCreateSecurity(v));
+                        })
+
+                        .section("amount") //
+                        .match(".*Net Cash Proceeds.*\\$(?<amount>[\\.,\\d]+)$")
+                        .assign((t, v) -> t.setAmount(asAmount((v.get("amount")))))
+
+                        .section("fee") //
+                        .optional() //
+                        .match(".*Total Fees\\s+\\$(?<fee>[\\.,\\d]+)").assign((t, v) -> {
+                            t.getPortfolioTransaction().addUnit(
+                                            new Unit(Unit.Type.FEE, Money.of(asCurrencyCode(asCurrencyCode("USD")),
+                                                            asAmount((v.get("fee"))))));
+                        })
+
+                        .wrap(BuySellEntryItem::new));
+    }
+
+    @Override
+    public String getLabel()
+    {
+        return "Fidelity"; //$NON-NLS-1$
+    }
+
+    @Override
+    protected long asAmount(String value)
+    {
+        String language = "de";
+        String country = "DE";
+
+        int lastDot = value.lastIndexOf(".");
+        int lastComma = value.lastIndexOf(",");
+
+        // returns the greater of two int values
+        if (Math.max(lastDot, lastComma) == lastDot)
+        {
+            language = "en";
+            country = "US";
+        }
+
+        return ExtractorUtils.convertToNumberLong(value, Values.Amount, language, country);
+    }
+
+    @Override
+    protected long asShares(String value)
+    {
+        String language = "de";
+        String country = "DE";
+
+        int lastDot = value.lastIndexOf(".");
+        int lastComma = value.lastIndexOf(",");
+
+        // returns the greater of two int values
+        if (Math.max(lastDot, lastComma) == lastDot)
+        {
+            language = "en";
+            country = "US";
+        }
```

Comment:
if all "asShares" are in en_US you can remove this and use
.assign((t, v) -> t.setShares(asShares(v.get("shares"), "en", "US"))),

The same for "asAmount"... 👍🏻 ...